OpenRA.Mods.Common/Traits/World/WeatherOverlay.cs

```diff
@@ -0,0 +1,391 @@
+				tempParticle.SwingDirection = !tempParticle.SwingDirection;
+
+			// Right direction turn
+			else if (tempParticle.SwingOffset > tempParticle.SwingAmplitude)
+				tempParticle.SwingDirection = !tempParticle.SwingDirection;
+
+			// Performe the X-Axis-Swing
+			if (tempParticle.SwingDirection)
+				tempParticle.SwingOffset += tempParticle.SwingSpeed;
+			else
+				tempParticle.SwingOffset -= tempParticle.SwingSpeed;
+
+			return tempParticle;
+		}
+
+		public void Tick(Actor self)
+		{
+			windTickCountdown--;
+		}
+
+		Particle WindLogic(Particle tempParticle)
+		{
+			if (info.ChangingWind)
+			{
+				if (windTickCountdown <= 0)
+				{
+					windTickCountdown = Game.CosmeticRandom.Next(info.WindTick[0], info.WindTick[1]);
+					if (Game.CosmeticRandom.Next(0, 2) == 1)
+					{
+						if (currentWindIndex > 0)
+						{
+							currentWindIndex--;
+							targetWindXOffset = info.WindStrength[currentWindIndex];
+						}
+					}
+					else
+					{
+						if (currentWindIndex < info.WindStrength.Length - 1)
+						{
+							currentWindIndex++;
+							targetWindXOffset = info.WindStrength[currentWindIndex];
+						}
+					}
+				}
+			}
+			else
+				targetWindXOffset = info.WindStrength[0];
+
+			// Fading the wind in little steps towards the TargetWindOffset
+			if (!info.InstantWindChanges)
+			{
+				if (currentWindXOffset != targetWindXOffset)
+				{
+					if (currentWindXOffset > targetWindXOffset)
+						currentWindXOffset -= 0.00001f;
+					else if (currentWindXOffset < targetWindXOffset)
+						currentWindXOffset += 0.00001f;
+				}
+			}
+			else { currentWindXOffset = targetWindXOffset; }
+
+			return tempParticle;
+		}
+
+		Particle Movement(Particle tempParticle)
+		{
+			tempParticle.PosX = tempParticle.PosX + tempParticle.DirectionScatterX + tempParticle.SwingOffset + currentWindXOffset;
+			tempParticle.PosY = tempParticle.PosY + tempParticle.Gravity;
+			return tempParticle;
+		}
+
+		// AntiScroll keeps the particels in place when scrolling the viewport
+		Particle AntiScroll(Particle tempParticle, WorldRenderer wr)
+		{
+			if (antiScrollPrevTopLeft != wr.Viewport.TopLeft) {
+				if (antiScrollPrevTopLeft.X != wr.Viewport.TopLeft.X) {
+					if (antiScrollPrevTopLeft.X > wr.Viewport.TopLeft.X)
+						tempParticle.PosX += antiScrollPrevTopLeft.X - wr.Viewport.TopLeft.X;
+					else
+						tempParticle.PosX -= wr.Viewport.TopLeft.X - antiScrollPrevTopLeft.X;
+				}
+
+				if (antiScrollPrevTopLeft.Y != wr.Viewport.TopLeft.Y) {
+					if (antiScrollPrevTopLeft.Y > wr.Viewport.TopLeft.Y)
+						tempParticle.PosY += antiScrollPrevTopLeft.Y - wr.Viewport.TopLeft.Y;
+					else
+						tempParticle.PosY -= wr.Viewport.TopLeft.Y - antiScrollPrevTopLeft.Y;
+				}
+			}
+
+			return tempParticle;
+		}
+
+		Particle EdgeCheckReplace(Particle tempParticle, WorldRenderer wr)
+		{
+			// X-Axis right
+			if (tempParticle.PosX > Game.Renderer.Resolution.Width)
+				tempParticle.PosX -= Game.Renderer.Resolution.Width;
```

Comment:
You should be able to replace this and the if below it with `tempParticle.PosX %= Game.Renderer.Resolution.Width`.  You can then do the same for height.